src/components/List.js

```diff
@@ -1,39 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { FirestoreDocument } from 'react-firestore';
-import ListItem from './ListItem';
-import '../css/components/ItemsList.css';
+import SortedList from './SortedList';
 import { getToken } from '../lib/TokenService';
 import AddButton from './AddButton';
+import { filter as removePunctuation } from '../lib/helper';
+import '../css/components/ItemsList.css';
 
 export default function List() {
+  const [searchTerm, setSearchTerm] = useState('');
+
   return (
     <div>
-      <h1>LIST OF ITEMS</h1>
+      <h3>LIST OF ITEMS</h3>
       <FirestoreDocument
         path={`shoppingLists/${getToken()}`}
         render={({ isLoading, data }) => {
-          const itemsKeys = data
-            ? Object.keys(data).filter((key) => key !== 'id')
-            : '';
+          let itemsKeys = data ? getItemsKeys(data, searchTerm) : [];
           return isLoading ? (
             <div className="m-auto">Loading</div>
           ) : (
             <div>
-              {itemsKeys.length >= 1 ? (
-                <ul>
-                  {itemsKeys
-                    .sort((a, b) => (a > b ? 1 : -1))
-                    .map((key) => {
-                      return (
-                        <ListItem
-                          listItem={data[key]}
-                          key={key}
-                          listId={data.id}
-                          itemId={key}
-                        />
-                      );
-                    })}
-                </ul>
+              {itemsKeys.length || searchTerm ? (
```

Comment:
Good Job here 👍
you can also add another check if `searchTerm && itemsKeys.length === 0` to view a message to the user if non of the items matches the search term

Reply:
I think that's a good idea! Thank you, Sarah! :)

Reply:
Great suggestion here Sarah.